### PR TITLE
cleaner rpcs

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this cleans up some rpc gen stuff, long overdue. decided to spend an hour on it. now we don't need to pass in the method const or have the params and call engine.rpc etc

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 1255e536f0ef...c64d6968a24a introduced some visual changes on win32:
 * Changed: **user_search-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468424410006-1255e536f0ef-c64d6968a24a-win32/user_search-Normal-1255e536f0ef.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468424410006-1255e536f0ef-c64d6968a24a-win32/user_search-Normal-c64d6968a24a.png)  
   user_search-Normal comparison
 * Changed: **unlock_folders-Normal** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468424410006-1255e536f0ef-c64d6968a24a-win32/unlock_folders-Normal-1255e536f0ef.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468424410006-1255e536f0ef-c64d6968a24a-win32/unlock_folders-Normal-c64d6968a24a.png)  
   unlock_folders-Normal comparison

File: protocol/bin/flow.js [hunk 1]
```diff
@@ -224,11 +241,10 @@ export type RPCError = {
   code: number,
   desc: string
 }
-
 `
 
-  const incomingMap = `export type incomingCallMapType = {\n` +
-  Object.keys(project.incomingMaps).map(im => `  '${im}'?: ${project.incomingMaps[im]}`).join(',\n') + '\n}\n\n'
+  const incomingMap = `export type incomingCallMapType = $Exact<{\n` +
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: also enforces no random keys in the incomingCallMap

**Comment by MarcoPolo** [comment]
Can we split this into parts to review and parts that are autogenerated?

**Comment by chrisnojima** [comment]
@MarcoPolo here https://github.com/keybase/client/pull/3456 for reference

File: protocol/bin/flow.js [hunk 2]
```diff
@@ -198,7 +200,18 @@ function parseRecord (t) {
...
+      .filter(t => t)
+      .reduce((acc, t) => {
+        const clean = t[0].trim()
+        return acc.indexOf(clean) === -1 ? acc.concat([clean]) : acc
+      }, [])
+      .filter(t => t)
+      .sort()
+      .join('\n  | ')
+
+  console.log(rpcTypes)
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: - [ ] remove

**Comment by MarcoPolo** [comment]
👍

**Pull request merged**